### PR TITLE
Migrate Travis CI to Github Actions

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Migrate Travis CI to Github Actions
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | N/A
| How to test?  | Github Actions are :green_circle:  https://github.com/Progi1984/ps_mainmenu/actions

### CODE REVIEW

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            - name: Run PHP-CS-Fixer
+              run: ./vendor/bin/php-cs-fixer fix --dry-run --diff --using-cache=no --diff-format udiff
+
+    # Run PHPStan against the module and a PrestaShop release
+    phpstan:
+        name: PHPStan
+        runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                presta-versions: ['1.7.1.2', '1.7.2.5', '1.7.3.4', '1.7.4.4', '1.7.5.1', '1.7.6.9', '1.7.7.3', 'latest']
```
Comment (PierreRambaud) [inline .github/workflows/php.yml:52]:
```suggestion
                presta-versions: ['1.7.1.2', '1.7.2.5', '1.7.3.4', '1.7.4.4', '1.7.5.1', '1.7.6.9', '1.7.7.4', 'latest']
```
no?

File: .php_cs.dist [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+$config = new PrestaShop\CodingStandards\CsFixer\Config();
+
+$config
+    ->setUsingCache(true)
+    ->getFinder()
+    ->in(__DIR__)
+    ->exclude('vendor');
+
+return $config;
```
Comment (PierreRambaud) [inline .php_cs.dist:11]:
```suggestion
return $config;

```

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
 /config_*.xml
 /translations/*.php
 /vendor/
+/.php_cs.cache
```
Comment (PierreRambaud) [inline .gitignore:4]:
```suggestion
/.php_cs.cache

```

File: tests/phpstan/phpstan-1.7.1.2.neon [hunk 4]
```diff
@@ -0,0 +1,2 @@
+includes:
+	- %currentWorkingDirectory%/tests/phpstan/phpstan.neon
```
Comment (PierreRambaud) [inline tests/phpstan/phpstan-1.7.1.2.neon:2]:
```suggestion
	- %currentWorkingDirectory%/tests/phpstan/phpstan.neon

```

File: tests/phpstan/phpstan-1.7.2.5.neon [hunk 5]
```diff
@@ -0,0 +1,2 @@
+includes:
+	- %currentWorkingDirectory%/tests/phpstan/phpstan.neon
```
Comment (PierreRambaud) [inline tests/phpstan/phpstan-1.7.2.5.neon:2]:
```suggestion
	- %currentWorkingDirectory%/tests/phpstan/phpstan.neon

```

File: tests/phpstan/phpstan.neon [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+parameters:
+  paths:
+    # From PHPStan 0.12, paths to check are relative to the neon file
+    - ../../ps_mainmenu.php
+    - ../../ps_menutoplinks.class.php
+    - ../../translations/
+    - ../../views/
+  reportUnmatchedIgnoredErrors: false
+  level: 5
```
Comment (PierreRambaud) [inline tests/phpstan/phpstan.neon:12]:
```suggestion
  level: 5

```

File: tests/phpstan/phpstan-latest.neon [hunk 7]
```diff
@@ -0,0 +1,2 @@
+includes:
+	- %currentWorkingDirectory%/tests/phpstan/phpstan.neon
```
Comment (PierreRambaud) [inline tests/phpstan/phpstan-latest.neon:2]:
```suggestion
	- %currentWorkingDirectory%/tests/phpstan/phpstan.neon

```

File: tests/phpstan/phpstan-1.7.7.3.neon [hunk 8]
```diff
@@ -0,0 +1,2 @@
+includes:
+	- %currentWorkingDirectory%/tests/phpstan/phpstan.neon
```
Comment (PierreRambaud) [inline tests/phpstan/phpstan-1.7.7.3.neon:2]:
```suggestion
	- %currentWorkingDirectory%/tests/phpstan/phpstan.neon

```

File: tests/phpstan/phpstan-1.7.6.9.neon [hunk 9]
```diff
@@ -0,0 +1,2 @@
+includes:
+	- %currentWorkingDirectory%/tests/phpstan/phpstan.neon
```
Comment (PierreRambaud) [inline tests/phpstan/phpstan-1.7.6.9.neon:2]:
```suggestion
	- %currentWorkingDirectory%/tests/phpstan/phpstan.neon

```

File: tests/phpstan/phpstan-1.7.5.1.neon [hunk 10]
```diff
@@ -0,0 +1,2 @@
+includes:
+	- %currentWorkingDirectory%/tests/phpstan/phpstan.neon
```
Comment (PierreRambaud) [inline tests/phpstan/phpstan-1.7.5.1.neon:2]:
```suggestion
	- %currentWorkingDirectory%/tests/phpstan/phpstan.neon

```

File: tests/phpstan/phpstan-1.7.4.4.neon [hunk 11]
```diff
@@ -0,0 +1,2 @@
+includes:
+	- %currentWorkingDirectory%/tests/phpstan/phpstan.neon
```
Comment (PierreRambaud) [inline tests/phpstan/phpstan-1.7.4.4.neon:2]:
```suggestion
	- %currentWorkingDirectory%/tests/phpstan/phpstan.neon

```

File: tests/phpstan/phpstan-1.7.3.4.neon [hunk 12]
```diff
@@ -0,0 +1,2 @@
+includes:
+	- %currentWorkingDirectory%/tests/phpstan/phpstan.neon
```
Comment (PierreRambaud) [inline tests/phpstan/phpstan-1.7.3.4.neon:2]:
```suggestion
	- %currentWorkingDirectory%/tests/phpstan/phpstan.neon

```

File: .github/workflows/build-release.yml [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2.0.0
+            - name: Install composer dependencies
+              run: composer install --no-dev -o
+            - name: Clean-up project
+              uses: PrestaShopCorp/github-action-clean-before-deploy@v1.0
+            - name: Prepare auto-index tool
+              run: |
+                  composer global require prestashop/autoindex
```
Comment (atomiix) [inline .github/workflows/build-release.yml:17]:
```suggestion
              run: composer global require prestashop/autoindex
```

File: .github/workflows/build-release.yml [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+            - name: Generate index.php
+              run: |
+                  ~/.composer/vendor/bin/autoindex
```
Comment (atomiix) [inline .github/workflows/build-release.yml:20]:
```suggestion
              run: ~/.composer/vendor/bin/autoindex
```

**Review by atomiix — DISMISSED** [review]
2 optional feedbacks ;)

File: .github/workflows/build-release.yml (hunk 13 above)
Comment (Progi1984) [reply .github/workflows/build-release.yml]: Done. Could you reapprove ?

File: .github/workflows/build-release.yml (hunk 14 above)
Comment (Progi1984) [reply .github/workflows/build-release.yml]: Done. Could you reapprove ?

**Review by PierreRambaud — DISMISSED** [review]

**Comment by Progi1984** [comment]
> 2 optional feedbacks ;)

@atomiix Done

**Review by PierreRambaud — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Comment by atomiix** [comment]
Thanks @Progi1984

**Pull request merged**